x/supplier/keeper/prune_supplier_service_config_history.go

```diff
@@ -0,0 +1,75 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// EndBlockerPruneSupplierServiceConfigHistory prunes the service config history
+// of suppliers that have service config history entries that are no longer needed
+// for pending claims settlement.
+func (k Keeper) EndBlockerPruneSupplierServiceConfigHistory(
+	ctx context.Context,
+) (numSuppliersWithPrunedHistory uint64, err error) {
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+	// The number of blocks from the end of a session to the end of the proof window close.
+	// It is needed to determine how long to retain service config updates for pending claims settlement.
+	sessionEndToProofWindowCloseNumBlocks := sharedtypes.GetSessionEndToProofWindowCloseBlocks(&sharedParams)
+
+	logger := k.Logger().With("method", "PruneSupplierServiceConfigHistory")
+
+	for _, supplier := range k.GetAllSuppliers(ctx) {
+		// Store the original history length for logging purposes.
```

Comment:
You use it for more than logging below.

```suggestion
		// Store the original number of historical service configs.
```